File: Backup/PlaceYourBets/FixtureBuilder.vb
Imports System.Net
Imports Newtonsoft.Json

Public Class FixtureBuilder

    Public Function getFixtures()
        Dim client = New WebClient()
        Dim reply = client.DownloadString("http://www.predictresults.co.uk/pradeep/SamH.php")

        'Dim fileReader As String
        'fileReader = My.Computer.FileSystem.ReadAllText("C:\Users\hamptons\Google Drive\PlaceYourBets\JSONexample - fixtures.txt")
        'Dim f = JsonConvert.DeserializeObject(Of FixtureList)(fileReader)

        Dim f = JsonConvert.DeserializeObject(Of FixtureList)(reply)

        Return f.stock
    End Function

    Public Function getBet()
        Dim client = New WebClient()
        Dim reply = client.DownloadString("http://www.predictresults.co.uk/pradeep/PYBPredictions.php")

        'Dim fileReader As String
        'fileReader = My.Computer.FileSystem.ReadAllText("C:\Users\hamptons\Google Drive\PlaceYourBets\JSONexample - predictions.txt")
        'Dim f = JsonConvert.DeserializeObject(Of FixtureList)(fileReader)

        Dim f = JsonConvert.DeserializeObject(Of FixtureList)(reply)

        Return f.stock
    End Function

    Public Function getUsers()
        Dim client = New WebClient
        Dim reply = client.DownloadString("http://www.predictresults.co.uk/pradeep/PYBUsers.php")
        Dim f = JsonConvert.DeserializeObject(Of UserList)(reply)

        'Dim fileReader As String
        'fileReader = My.Computer.FileSystem.ReadAllText("C:\Users\hamptons\Google Drive\PlaceYourBets\JSONexample - users.txt")
        'Dim f = JsonConvert.DeserializeObject(Of UserList)(fileReader)

        Dim list As New List(Of String)

        For Each user As Users In f.stock
            list.Add(user.user)
        Next

        Return list

    End Function

    Public Function getScores()
        Dim client = New WebClient
        Dim reply = client.DownloadString("http://www.predictresults.co.uk/pradeep/UserStats.php")
        Dim f = JsonConvert.DeserializeObject(Of UserList)(reply)

        'Dim fileReader As String
        'fileReader = My.Computer.FileSystem.ReadAllText("C:\Users\hamptons\Google Drive\PlaceYourBets\JSONexample - users.txt")
        'Dim f = JsonConvert.DeserializeObject(Of UserList)(fileReader)

        Return f.stock
    End Function

    Public Function usersSubmitted()
        Dim client = New WebClient
        Dim reply = client.DownloadString("http://www.predictresults.co.uk/pradeep/WhoBet.php")
        Dim f = JsonConvert.DeserializeObject(Of UserList)(reply)

        Dim list As New List(Of String)

        For Each user As Users In f.stock
            list.Add(user.user)
        Next

        Return list

    End Function

    Public Function getUsersAndIds()
        Dim client = New WebClient
        Dim reply = client.DownloadString("http://www.predictresults.co.uk/pradeep/PYBUsers.php")
        Dim f = JsonConvert.DeserializeObject(Of UserList)(reply)

        Dim dictionary As New Dictionary(Of String, Integer)

        For Each user As Users In f.stock
            dictionary.Add(user.user, user.UserID)
        Next

        Return dictionary

    End Function

    Public Function getIds()
        Dim client = New WebClient
        Dim reply = client.DownloadString("http://www.predictresults.co.uk/pradeep/PYBUsers.php")
        Dim f = JsonConvert.DeserializeObject(Of UserList)(reply)

        Dim list As New List(Of Integer)

        For Each user As Users In f.stock
            list.Add(user.UserID)
        Next

        Return list

    End Function

    Public Function getPRUsers()
        Dim client = New WebClient
        Dim reply = client.DownloadString("http://www.predictresults.co.uk/pradeep/PRUsers.php")
        Dim f = JsonConvert.DeserializeObject(Of UserList)(reply)

        Dim list As New List(Of String)

        For Each user As Users In f.stock
            list.Add(user.user)
        Next

        Return list
    End Function



End Class

File: PlaceYourBets.ConvertedToC#/FixtureBuilder.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;
using System.Net;
using Newtonsoft.Json;
namespace PlaceYourBets
{

	public class FixtureBuilder
	{

		public object getFixtures()
		{
			var client = new WebClient();
			var reply = client.DownloadString("http://www.predictresults.co.uk/API/SamH.php");

			//Dim fileReader As String
			//fileReader = My.Computer.FileSystem.ReadAllText("C:\Users\hamptons\Google Drive\PlaceYourBets\JSONexample - fixtures.txt")
			//Dim f = JsonConvert.DeserializeObject(Of FixtureList)(fileReader)

			var f = JsonConvert.DeserializeObject<FixtureList>(reply);

			return f.stock;
		}

		public object getBet()
		{
			var client = new WebClient();
			var reply = client.DownloadString("http://www.predictresults.co.uk/API/PYBPredictions.php");

			//Dim fileReader As String
			//fileReader = My.Computer.FileSystem.ReadAllText("C:\Users\hamptons\Google Drive\PlaceYourBets\JSONexample - predictions.txt")
			//Dim f = JsonConvert.DeserializeObject(Of FixtureList)(fileReader)

			var f = JsonConvert.DeserializeObject<FixtureList>(reply);

			return f.stock;
		}

		public object getUsers()
		{
			var client = new WebClient();
			var reply = client.DownloadString("http://www.predictresults.co.uk/API/PYBUsers.php");
			var f = JsonConvert.DeserializeObject<UserList>(reply);

			//Dim fileReader As String
			//fileReader = My.Computer.FileSystem.ReadAllText("C:\Users\hamptons\Google Drive\PlaceYourBets\JSONexample - users.txt")
			//Dim f = JsonConvert.DeserializeObject(Of UserList)(fileReader)

			List<string> list = new List<string>();

			foreach (Users user in f.stock) {
				list.Add(user.user);
			}

			return list;

		}

		public object getScores()
		{
			var client = new WebClient();
			var reply = client.DownloadString("http://www.predictresults.co.uk/API/UserStats.php");
			var f = JsonConvert.DeserializeObject<UserList>(reply);

			//Dim fileReader As String
			//fileReader = My.Computer.FileSystem.ReadAllText("C:\Users\hamptons\Google Drive\PlaceYourBets\JSONexample - users.txt")
			//Dim f = JsonConvert.DeserializeObject(Of UserList)(fileReader)

			return f.stock;
		}

		public object usersSubmitted()
		{
			var client = new WebClient();
			var reply = client.DownloadString("http://www.predictresults.co.uk/API/WhoBet.php");
			var f = JsonConvert.DeserializeObject<UserList>(reply);

			List<string> list = new List<string>();


			try {
				foreach (Users user in f.stock) {
					list.Add(user.user);
				}


			} catch (NullReferenceException ex) {
			}



			return list;

		}

		public object getUsersAndIds()
		{
			var client = new WebClient();
			var reply = client.DownloadString("http://www.predictresults.co.uk/API/PYBUsers.php");
			var f = JsonConvert.DeserializeObject<UserList>(reply);

			Dictionary<string, int> dictionary = new Dictionary<string, int>();

			foreach (Users user in f.stock) {
				dictionary.Add(user.user.ToLower(), user.UserID);
			}

			return dictionary;

		}

		public object getIdsandUsers()
		{
			var client = new WebClient();
			var reply = client.DownloadString("http://www.predictresults.co.uk/API/PYBUsers.php");
			var f = JsonConvert.DeserializeObject<UserList>(reply);

			Dictionary<int, string> dictionary = new Dictionary<int, string>();

			foreach (Users user in f.stock) {
				dictionary.Add(user.UserID, user.user);
			}

			return dictionary;

		}

		public object getIds()
		{
			var client = new WebClient();
			var reply = client.DownloadString("http://www.predictresults.co.uk/API/PYBUsers.php");
			var f = JsonConvert.DeserializeObject<UserList>(reply);

			List<int> list = new List<int>();

			foreach (Users user in f.stock) {
				list.Add(user.UserID);
			}

			return list;

		}

		public object getPRUsers()
		{
			var client = new WebClient();
			var reply = client.DownloadString("http://www.predictresults.co.uk/API/PRUsers.php");
			var f = JsonConvert.DeserializeObject<UserList>(reply);

			List<string> list = new List<string>();

			foreach (Users user in f.stock) {
				list.Add(user.user);
			}

			return list;
		}

	}
}
